jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/Validator.java

```diff
@@ -27,35 +28,116 @@
 public class Validator {
 
   /**
-   * Checks if string is null, empty or only whitespace.
+   * Checks if string is non null and non empty.
    *
    * @param value the string in question
    * @param propertyName the equivalent 'yaml' property name
    * @throws NullPointerException if {@code value} is null
    * @throws IllegalArgumentException if {@code value} is empty or only whitespace
    */
-  public static void checkNotEmpty(@Nullable String value, String propertyName) {
+  public static void checkNotNullAndNotEmpty(@Nullable String value, String propertyName) {
     Preconditions.checkNotNull(value, "Property '" + propertyName + "' cannot be null");
     Preconditions.checkArgument(
         !value.trim().isEmpty(), "Property '" + propertyName + "' cannot be empty");
   }
 
   /**
-   * Checks if a collection is null or empty.
+   * Checks if string is null or non empty.
+   *
+   * @param value the string in question
+   * @param propertyName the equivalent 'yaml' property name
+   * @throws IllegalArgumentException if {@code value} is empty or only whitespace
+   */
+  public static void checkNullOrNotEmpty(@Nullable String value, String propertyName) {
+    if (value == null) {
+      // pass
+      return;
+    }
+    Preconditions.checkArgument(
+        !value.trim().isEmpty(), "Property '" + propertyName + "' cannot be empty");
+  }
+
+  /**
+   * Checks if a collection is not null and not empty.
    *
    * @param value the string in question
    * @param propertyName the equivalent 'yaml' property name
    * @throws NullPointerException if {@code value} is null
    * @throws IllegalArgumentException if {@code value} is empty
    */
-  public static void checkNotEmpty(@Nullable Collection<?> value, String propertyName) {
+  public static void checkNotNullAndNotEmpty(@Nullable Collection<?> value, String propertyName) {
```

Comment:
Remove `@Nullable`? Or, does it cause some compilation trouble with NullAway?

Reply:
It technically accepts null values. Removing `@Nullable` would mean a check to use this method.?

Reply:
Ah, I guess `@Nullable` makes sense as a check method.